packages/web3-core-requestmanager/src/index.js

```diff
@@ -123,11 +123,27 @@ RequestManager.prototype.setProvider = function (provider, net) {
             });
         });
 
-        // notify all subscriptions about the close condition
+        // notify all subscriptions about bad close conditions
         this.provider.on('close', function close(event) {
-            _this.subscriptions.forEach(function (subscription) {
-                subscription.callback(new Error('CONNECTION ERROR: The connection got closed with the close code `' + event.code + '` and the following reason string `' + event.reason + '`'));
-            });
+            // The node.net module emits ('close', isException)
+            const ipcCloseError = typeof event === 'boolean' && event;
+
+            // The WS package has a close event w/ codes & reasons
+            const hasErrorCode = ![1000, 1001].includes(event.code);
+
+            if (ipcCloseError || hasErrorCode || event.wasClean === false){
+                _this.subscriptions.forEach(function (subscription) {
+                    const message = (hasErrorCode)
```

Comment:
We could move the error message logic out to the ``web3-core-helpers`` package with his error functions. This would clean up those lines a bit.